File: app/src/main/java/com/wojdor/hearthcards/application/util/JsonParser.java
package com.wojdor.hearthcards.application.util;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

import timber.log.Timber;

public class JsonParser {

    private Gson gson = new Gson();

    public <T> String toJson(T object) {
        return gson.toJson(object);
    }

    public <T> T fromJson(Class<T> type, String json) {
        T object = null;
        try {
            object = gson.fromJson(json, type);
        } catch (JsonSyntaxException error) {
            Timber.e(error);
        }
        return object;
    }
}

File: app/src/main/java/com/wojdor/hearthstonecards/app/util/JsonParser.kt
package com.wojdor.hearthstonecards.app.util

import com.google.gson.Gson
import com.google.gson.JsonSyntaxException

import timber.log.Timber

class JsonParser {

    private val gson = Gson()

    fun <T> toJson(objectToParse: T): String {
        return gson.toJson(objectToParse)
    }

    fun <T> fromJson(type: Class<T>?, json: String): T? {
        var parsedObject: T? = null
        try {
            parsedObject = gson.fromJson(json, type)
        } catch (error: JsonSyntaxException) {
            Timber.e(error)
        }
        return parsedObject
    }
}
